### PR TITLE
@dleve123 => [Sale] Be more defensive about fetching pages > 100 during inf. scroll

### PR DESCRIPTION
**Pull request by mzikherman**
Re-opening of https://github.com/artsy/force/pull/4693 (more info there) but from an upstream branch instead of a fork (I think there's some secrets not passed to forks by Circle that are now needed for Force CI).

### CODE REVIEW

File: src/desktop/apps/auction/reducers/__tests__/artworkBrowser.test.js [hunk 1]
```diff
@@ -153,6 +153,22 @@ describe("auction/actions/artworkBrowser.test.js", () => {
...
+            artworkBrowser: {
+              total: 1,
+              saleArtworks: [],
+              filterParams: { page: 100 },
+            },
+          },
+          {}
+        )
+        const tooHighPage = auctions(maxPageResponse, actions.updatePage(false))
+        const notAllFetched = auctions(tooHighPage, actions.updateAllFetched())
```
Comment (dleve123) [inline src/desktop/apps/auction/reducers/__tests__/artworkBrowser.test.js]: Do we need to explicitly pass on `false` like we do above? Is our goal here to stub redux state? Just trying to make sure I understand the test

File: src/desktop/apps/auction/reducers/artworkBrowser.js [hunk 2]
```diff
@@ -105,7 +105,8 @@ export default function auctionArtworkFilter(state = initialState, action) {
       )
     }
     case actions.UPDATE_ALL_FETCHED: {
-      if (state.saleArtworks.length === state.total) {
+      const currentPage = state.filterParams.page
+      if (currentPage > 100 || state.saleArtworks.length === state.total) {
```
Comment (dleve123) [inline src/desktop/apps/auction/reducers/artworkBrowser.js]: So if I understand correctly this limits us to 100*25 (results/page) - 2,500 artworks/sale? If so, this seems reasonable to me - just want to make sure I understand.

Comment (dleve123) [reply src/desktop/apps/auction/reducers/artworkBrowser.js]: Log line in https://artsy.slack.com/archives/CA8SANW3W/p1571265677276700 is what I’m basing the 25 number off of btw

File: src/desktop/apps/auction/reducers/__tests__/artworkBrowser.test.js (hunk 1 above)
Comment (mzikherman) [reply src/desktop/apps/auction/reducers/__tests__/artworkBrowser.test.js]: Yea, the `false` param for `updatePage` is what tells that API to increment the page, vs setting it to 1. It's a 'reset' argument: https://github.com/artsy/force/blob/cc12124d1b1982eac2a9d6b571d798a830ae55ac/src/desktop/apps/auction/reducers/artworkBrowser.js#L255

File: src/desktop/apps/auction/reducers/artworkBrowser.js (hunk 2 above)
Comment (mzikherman) [reply src/desktop/apps/auction/reducers/artworkBrowser.js]: It's actually 20 results/page, not 25. The page size can be increased if needed, but the max page of 100 is enforced at the API level, so this enforces that at the client level here too. Always better to be more defensive about that stuff when dealing with an inf. scroll situation.

Comment (mzikherman) [reply src/desktop/apps/auction/reducers/artworkBrowser.js]: That's set in https://github.com/artsy/force/blob/cc12124d1b1982eac2a9d6b571d798a830ae55ac/src/desktop/apps/auction/reducers/artworkBrowser.js#L20 (also reflected in the log line).

File: src/desktop/apps/auction/reducers/__tests__/artworkBrowser.test.js (hunk 1 above)
Comment (dleve123) [reply src/desktop/apps/auction/reducers/__tests__/artworkBrowser.test.js]: gotcha, and the `const notAllFetched = auctions(tooHighPage, actions.updateAllFetched())` is the code actually being exercised in the test

Comment (dleve123) [reply src/desktop/apps/auction/reducers/__tests__/artworkBrowser.test.js]: So just to be super clear, the `const tooHighPage = auctions(maxPageResponse, actions.updatePage(false))` gets us to the `101`st page, hence triggering the  `currentPage > 100` that you added, right?

**Pull request merged**